src/org/jitsi/impl/neomedia/RTPConnectorInputStream.java

```diff
@@ -743,9 +743,9 @@ private void runInReceiveThread()
 
         while (!closed)
         {
-            // http://code.google.com/p/android/issues/detail?id=24765
-            if (OSUtils.IS_ANDROID)
-                p.setLength(PACKET_RECEIVE_BUFFER_LENGTH);
+            // Reset the buffer, because the previous call to receive() might
+            // have bumped the offset or even changed the byte[].
+            p.setData(buffer, 0, buffer.length);
```

Comment:
> the octo logic won't always kick in, will it?
Correct.

> does it make sense to have the reset (in the case of stripping the octo buffers) in the octo receive logic so its transparent elsewhere?

I don't think this is possible, because the reset() needs to happen after the packet has been consumed by libjitsi. The flow is like this:
1. Libjitsi calls receive() with a buffer that it manages
2. The receive goes down to underlying sockets which copies data to the packet's buffer, at the packet's offset
3. Some Octo logic (part of the overriden DatagramSocket#receive()) reads the first 8 bytes and bumps the offset by 8
4. Libjitsi handles the received packet
5. Libjitsi calls receive() again

The reset happens between 4 and 5.



> if i'm understanding what you said correctly (i don't think i am), how is the octo receive bumping the offset ahead different than the normal processing of a packet which reads the buffer?

The Octo logic basically does p.setOffset(p.getOffset()+8). When libjitsi reads the packet it doesn't modify the offset, AFAIK.